crates/dpapi/src/rpc/verification.rs

```diff
@@ -0,0 +1,259 @@
+    Bitmask1 = 0x0001,
+    Pcontext = 0x0002,
+    Header2 = 0x0003,
+}
+
+impl CommandType {
+    pub fn as_u16(self) -> u16 {
+        self as u16
+    }
+}
+
+bitflags! {
+    #[derive(Debug, Copy, Clone, PartialEq, Eq, Default)]
+    pub struct CommandFlags: u16 {
+        const None = 0;
+        const SecVtCommandEnd = 0x4000;
+        const SecVtMustProcessCommand = 0x8000;
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Command {
+    Bitmask1(CommandBitmask),
+    Pcontext(CommandPContext),
+    Header2(CommandHeader2),
+}
+
+impl Command {
+    pub fn flags(&self) -> CommandFlags {
+        match self {
+            Command::Bitmask1(command) => command.flags,
+            Command::Pcontext(command) => command.flags,
+            Command::Header2(command) => command.flags,
+        }
+    }
+
+    pub fn command_type(&self) -> CommandType {
+        match self {
+            Command::Bitmask1(_) => CommandType::Bitmask1,
+            Command::Pcontext(_) => CommandType::Pcontext,
+            Command::Header2(_) => CommandType::Header2,
+        }
+    }
+}
+
+impl Encode for Command {
+    fn encode(&self, mut writer: impl Write) -> DpapiResult<()> {
+        writer.write_u16::<LittleEndian>(self.command_type().as_u16() | self.flags().bits())?;
+
+        match self {
+            Command::Bitmask1(command) => command.encode_value(writer),
+            Command::Pcontext(command) => command.encode_value(writer),
+            Command::Header2(command) => command.encode_value(writer),
+        }
+    }
+}
+
+impl Decode for Command {
+    fn decode(mut reader: impl Read) -> DpapiResult<Self> {
+        let cmd_field = reader.read_u16::<LittleEndian>()?;
+
+        let command_type = cmd_field & 0x3fff;
+        let command_flags = cmd_field & 0xc000;
+
+        let command = CommandType::from_u16(command_type).ok_or(CommandError::InvalidCommandType(command_type))?;
+        let flags = CommandFlags::from_bits(command_flags).ok_or(CommandError::InvalidCommandFlags(command_flags))?;
+
+        let value_len = reader.read_u16::<LittleEndian>()?;
+        let value = read_vec(usize::from(value_len), reader)?;
+
+        Ok(match command {
+            CommandType::Bitmask1 => Self::Bitmask1(CommandBitmask::from_flags_and_value(flags, &value)?),
+            CommandType::Pcontext => Self::Pcontext(CommandPContext::from_flags_and_value(flags, &value)?),
+            CommandType::Header2 => Self::Header2(CommandHeader2::from_flags_and_value(flags, &value)?),
+        })
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct CommandBitmask {
+    pub bits: u32,
+    pub flags: CommandFlags,
+}
+
+impl CommandBitmask {
+    fn from_flags_and_value(flags: CommandFlags, value: &[u8]) -> DpapiResult<Self> {
+        if value.len() != 4 {
+            Err(CommandError::InvalidCommandBitmaskValueLength(value.len()))?;
+        }
+
+        let bits: [u8; 4] = value.try_into().expect("length should be checked");
```

Comment:
nitpick: A clearer wording is:
```suggestion
        let bits: [u8; 4] = value.try_into().expect("length is checked above");
```

Reply:
done